### ISSUE TITLE
Extend the Go Client's CreateProcessInstance API with start instructions

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#9399]
The Go client should implement the extended CreateProcessInstance API with start instructions.

Specifically, we'll need to add start instructions to the CreateInstanceCommandStep3, the API depends on the design choices made in https://github.com/camunda/zeebe/issues/9388. Let's be inspired by #9398.

Blocked by https://github.com/camunda/zeebe/issues/9388, https://github.com/camunda/zeebe/issues/9396

**Comment by korthout** [comment]
@saig0 We should verify whether `zbctl` is out of scope for this feature.

**Comment by saig0** [comment]
The relevance for `zbctl` is discussed here: https://camunda.slack.com/archives/C03C0HS85BL/p1655109212920579
